plugin/src/test/java/io/jenkins/plugins/analysis/warnings/steps/StepsITest.java

```diff
@@ -732,6 +734,63 @@ public void shouldShowWarningsOfGroovyParser() {
                 new GroovyParser(id, "Groovy Pep8",
                         "(.*):(\\d+):(\\d+): (\\D\\d*) (.*)",
                         toString("groovy/pep8.groovy"), "")));
+        testGroovyPep8JobIsSuccessful(job, id);
+    }
+
+    /**
+     * Registers a new {@link GroovyParser} (a Pep8 parser) in Jenkins global configuration and runs this parser on the console that's showing an
+     * error log with 8 issues.
+     * @throws IOException if the test fails unexpectedly
+     */
+    @Test
+    public void shouldShowWarningsOfGroovyParserWhenScanningConsoleLogWhenThatIsPermitted() throws IOException {
+        WorkflowJob job = createPipeline();
+        ArrayList<String> stages = new ArrayList<>();
+        catFileContentsByAddingEchosSteps(stages, "pep8Test.txt");
+        stages.add("def groovy = scanForIssues "
+                + "tool: groovyScript(parserId: 'groovy-pep8', pattern:'', reportEncoding:'UTF-8')");
+        stages.add("publishIssues issues:[groovy]");
+        job.setDefinition(asStage(stages.toArray(new String[0])));
+
+        ParserConfiguration configuration = ParserConfiguration.getInstance();
+        configuration.setConsoleLogScanningPermitted(true);
```

Comment:
CPD is complaining that it `Found duplicated code` and I don't intend to fix that - many unit tests are fairly repetitive by nature.
I've extracted some code out into common methods to reduce duplication but there will inevitably be some commonality between related tests.

Reply:
Yes, this is ok! There is no quality gate enforced for CPD anyway, it is just to provide some feedback.